runsc/sandbox/network.go

```diff
@@ -278,6 +287,28 @@ func createInterfacesAndRoutesFromNS(conn *urpc.Client, nsPath string, hardwareG
 	return nil
 }
 
+// isAddressOnInterface checks if an address is on an interface
+func isAddressOnInterface(ifaceName string, addr *net.IPNet) (bool, error) {
+	iface, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return false, fmt.Errorf("getting interface by name %q: %w", ifaceName, err)
+	}
+	ifaceAddrs, err := iface.Addrs()
+	if err != nil {
+		return false, fmt.Errorf("fetching interface addresses for %q: %w", iface.Name, err)
+	}
+	for _, ifaceAddr := range ifaceAddrs {
+		ipNet, ok := ifaceAddr.(*net.IPNet)
+		if !ok {
```

Comment:
Should it just skip the address in this case?

Reply:
I was copying the behavior of line 168 that returns an error if it finds an address on an interface it can't cast to `*net.IPNet`.
I agree with you that having a `continue` here would make this helper function more resilient so I'll go ahead and do that.